docs/guides/migration_guide_packet.md

```diff
@@ -0,0 +1,152 @@
+
+```hcl-terraform
+terraform {
+  required_providers {
+    packet = {
+      source = "packethost/packet"
+      version = "3.2.1"
+    }
+  }
+}
+
+
+resource "packet_reserved_ip_block" "example" {
+  project_id = local.project_id
+  facility   = "sv15"
+  quantity   = 2
+}
+
+resource "packet_device" "example" {
+  project_id       = local.project_id
+  facilities       = ["sv15"]
+  plan             = "c3.medium.x86"
+  operating_system = "ubuntu_20_04"
+  hostname         = "test"
+  billing_cycle    = "hourly"
+
+  ip_address {
+    type            = "public_ipv4"
+    cidr            = 31
+    reservation_ids = [packet_reserved_ip_block.example.id]
+  }
+
+  ip_address {
+    type = "private_ipv4"
+  }
+}
+```
+
+## Resource UUIDs
+
+In order to transition to provider equinix/metal, we need to find out UUIDs of all the resources we want to migrate. In this case `packet_reserved_ip_block.example` and `packet_device.example`. We can use `terraform state` to find out the UUIDs.
+
+For the reserved IP block:
+
+```
+$ terraform state show packet_reserved_ip_block.example
+# packet_reserved_ip_block.example:
+resource "packet_reserved_ip_block" "example" {
+	[...]
+    id = "e689072f-aa6e-4d51-8e37-c2fbe18b4ff0"
+	[...]
+}
+```
+
+For the device:
+
+```
+$ terraform state show packet_device.example
+# packet_device.example
+resource "packet_device" "example" {
+	[...]
+    id = "8eb3bc10-0e1a-476a-aec2-6dc699df9c1c"
+	[...]
+
+```
+
+## Migrated template
+
+Once we find out the UUIDs of resources to migrate, in the HCL template, we need to change:
+ 
+* the required_providers block to require equinix/metal
+* the names of the resources to corresponding resoruces from provider quinix/etal (sed 's/packet_/metal_')
+* all the references to packet_ resources to metal_ resources
+
+The modified template will then look as
+
+```hcl-terraform
+terraform {
+  required_providers {
+    metal = {
+      source = "equinix/metal"
+      version = "2.0.1"
+    }
+  }
+}
+
+
+resource "metal_reserved_ip_block" "example" {
+  project_id = local.project_id
+  facility   = "sv15"
+  quantity   = 2
+}
+
+resource "metal_device" "example" {
+  project_id       = local.project_id
+  facilities       = ["sv15"]
+  plan             = "c3.medium.x86"
+  operating_system = "ubuntu_20_04"
+  hostname         = "test"
+  billing_cycle    = "hourly"
+
+  ip_address {
+    type            = "public_ipv4"
+    cidr            = 31
+    reservation_ids = [metal_reserved_ip_block.example.id]
+  }
+
+  ip_address {
+    type = "private_ipv4"
+  }
+}
+```
+
+## Migrating Terraform state
+
+Once we changed the template accordingly, we can remove the old packet_ resources from Terraform state and import the new ones as metal_ resources by their UUIDs.
+
+From checking the state before, we remember that UUID of the packet_device.xample is 8eb3bc10-0e1a-476a-aec2-6dc699df9c1c, and UUID of the packet_reserved_ip_block.example is e689072f-aa6e-4d51-8e37-c2fbe18b4ff0.
```

Comment:
I imagine they will have to run `terraform init -upgrade` somewhere in this guide.